### PR TITLE
Add RSA encrypt demo

### PR DESCRIPTION
**Pull request by slontis**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: demos/encrypt/rsa_encrypt.c [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+{
+    OSSL_DECODER_CTX *dctx = NULL;
+    EVP_PKEY  *pkey = NULL;
+    int selection;
+    const unsigned char *data;
+    size_t data_len;
+
+    if (public) {
+        selection = EVP_PKEY_PUBLIC_KEY;
+        data =  pub_key_der;
```
Comment (InfoHunter) [inline demos/encrypt/rsa_encrypt.c:46]: Nit: extra whitespace ahead of `pub_key_der`.

File: demos/encrypt/rsa_encrypt.c [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+        data_len = sizeof(pub_key_der);
+    } else {
+        selection =  EVP_PKEY_KEYPAIR;
```
Comment (InfoHunter) [inline demos/encrypt/rsa_encrypt.c:49]: Same nit here.

File: demos/README.txt [hunk 3]
```diff
@@ -18,6 +18,9 @@ EVP_MD_demo.c          Compute a digest from multiple buffers
 EVP_MD_stdin.c         Compute a digest with data read from stdin
 EVP_f_md.c             Compute a digest using BIO and EVP_f_md
 
+encrypt:
+rsa_encrypt.c          Encrypt and Decrypt data using an RSA keypair.
```
Comment (paulidale) [inline demos/README.txt:22]: Lower case "Decrypt" since it is mid sentence?

File: demos/encrypt/rsa_encrypt.c [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+    "Or to take Armes again in a sea of troubles";
+
+/*
+ * For do_encrypt(), load an RSA public key from pub_key_der[].
+ * For do_decrypt(), load an RSA private key from priv_key_der[].
+ */
+static EVP_PKEY *get_key(OSSL_LIB_CTX *libctx, const char *propq, int public)
+{
+    OSSL_DECODER_CTX *dctx = NULL;
+    EVP_PKEY  *pkey = NULL;
```
Comment (paulidale) [inline demos/encrypt/rsa_encrypt.c:39]: double space here it seems.

File: demos/encrypt/rsa_encrypt.c [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+                      const unsigned char *in, size_t in_len,
+                      unsigned char **out, size_t *out_len)
+{
+    int ret = 0, public = 1;
+    size_t buf_len = 0;
+    unsigned char *buf = NULL;
+    const char *propq = NULL;
+    EVP_PKEY_CTX *ctx = NULL;
+    EVP_PKEY *pub_key = NULL;
+    OSSL_PARAM params[8];
```
Comment (paulidale) [inline demos/encrypt/rsa_encrypt.c:106]: Why `8` here?

**Review by InfoHunter — APPROVED** [review]
It's not necessary to fix the nits

File: demos/README.txt (hunk 3 above)
Comment (InfoHunter) [reply demos/README.txt:22]: I also noticed this. But I think Shane might want to highlight the functionality of this demo ;-)

**Comment by InfoHunter** [comment]
>  It's not necessary to fix the nits

I didn't see Pauli's comments when sending this out...

**Comment by paulidale** [comment]
The NITs should be fixed before merging.

File: demos/encrypt/rsa_encrypt.c (hunk 5 above)
Comment (slontis) [reply demos/encrypt/rsa_encrypt.c:106]: Changed to 5

**Comment by slontis** [comment]
NITS fixed.

**Review by paulidale — APPROVED** [review]

**Comment by mattcaswell** [comment]
Pushed to master. Thanks.

**Pull request closed without merging**